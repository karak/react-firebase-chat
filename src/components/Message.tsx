import * as React from 'react';
import { Comment } from 'semantic-ui-react';

export interface MessageProps {
  speaker: string;
  speech: string;
  time: React.ReactNode;
}

export default class Message extends React.PureComponent<MessageProps> {
  render() {
    return (
      <Comment>
        <Comment.Avatar src="" />
        <Comment.Content>
          <Comment.Author as="a">{this.props.speaker}</Comment.Author>
          <Comment.Metadata>
            <div>{this.props.time}</div>
          </Comment.Metadata>
          <Comment.Text>{this.props.speech}</Comment.Text>
        </Comment.Content>
      </Comment>
    );
  }

  // tslint:disable-next-line:function-name
  static Group({ children }: { children?: React.ReactNode }) {
    return <Comment.Group>{children}</Comment.Group>;
  }
}
