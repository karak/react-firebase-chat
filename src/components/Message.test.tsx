import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Comment } from 'semantic-ui-react';
import Message from './Message';

describe('<Message>', () => {
  const element = <Message speaker="John" speech="Hello." time="1:00" />;
  const wrapper = shallow(element);

  it('has <Comment.Author> with speaker', () => {
    expect(wrapper.find(Comment.Author)).toBePresent();
    expect(wrapper.find(Comment.Author).dive()).toHaveText('John');
  });

  it('has <Comment.Text> with speech', () => {
    expect(wrapper.find(Comment.Text)).toBePresent();
    expect(wrapper.find(Comment.Text).dive()).toHaveText('Hello.');
  });

  it('has <Comment.Metadata> with time', () => {
    expect(wrapper.find(Comment.Metadata)).toBePresent();
    expect(wrapper.find(Comment.Metadata).dive()).toHaveText('1:00');
  });
});

describe('<Message.Group>', () => {
  it('is rendered as <Comment.Group>', () => {
    const wrapper = shallow(<Message.Group/>);

    expect(wrapper.type()).toBe(Comment.Group);
  });

  it('retains snapshot', () => {
    const tree = renderer.create(<Message.Group/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
