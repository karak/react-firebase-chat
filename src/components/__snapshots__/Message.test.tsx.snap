// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message.Group> retains snapshot 1`] = `
<div
  className="ui comments"
/>
`;
